hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -1711,6 +2126,46 @@ private AbfsPerfInfo startTracking(String callerName, String calleeName) {
     return new AbfsPerfInfo(abfsPerfTracker, callerName, calleeName);
   }
 
+  /**
+   * Search for a FileStatus corresponding to a RenamePending JSON file.
+   * @param fileStatuses array of fileStatus from which JSON file has to be searched.
+   * @return filestatus corresponding to RenamePending JSON file.
+   */
+  public FileStatus getRenamePendingFileStatus(final FileStatus[] fileStatuses) {
+    for (FileStatus fileStatus : fileStatuses) {
+      if (fileStatus.getPath().toUri().getPath().endsWith(SUFFIX)) {
+        return fileStatus;
+      }
+    }
+    return null;
+  }
+
+  /**
+   * For a given directory, returns back the fileStatus information for the
+   * RenamePending JSON file for the directory.
+   *
+   * @param fileStatus FileStatus object of the directory for which JSON file has
+   * to be searched.
+   * @param tracingContext TracingContext object for tracing the backend server calls
+   * for the operation.
+   * @throws IOException exception thrown from the call to {@link #getPathStatus(Path, TracingContext)}
+   * method.
+   */
+  public boolean getRenamePendingFileStatusInDirectory(final FileStatus fileStatus,
+      final TracingContext tracingContext) throws IOException {
+    try {
+      getFileStatus(
+          new Path(fileStatus.getPath().toUri().getPath() + SUFFIX),
+          tracingContext);
+      return true;
```

Comment:
Is the function supposed to return true in every scenario ?

Reply:
three things can happen:
1. blob is there: no exception: return true;
2. blob is not there: exception with 404: return false
3. some other exception: throw it.